### DISCUSSION TITLE
Building Nixie & REM Chem From ET 4: My Adventure and Experience

### DISCUSSION THREAD

**Discussion by FlaMike** [tagyoureit/nodejs-poolController#936] [General]
This is part 1 of a multi-part experience summary of my build.  I am posting it to first thank those who created Nixie and REM Chem,  inspired, supported, guided, and continue to go above and beyond to help me with this project even today.  Your generosity and assistance have been vital to my successes with this project.  I hope the completed post will assist others who take on similar journeys and in the best case, help them avoid mistakes & detours along the way.

Background

I learned about njs-PC about 2 years ago when my one year old ET 4 pooped out & lost communication & control abilities.  I had a wired Ethernet connection to the ET4 and assumed it would be superior and more reliable than the wireless option.  Silly me, as it turned out.

The ET lost its motherboard, possibly following a thunderstorm.  The house, of course, never lost power during the storm.  While still under warranty, it took more than 3 weeks before a replacement could be shipped and installed.  The board failure occurred in July and I was not a happy camper, to put it mildly.

I’ve been running Home Assistant (HA) for several years & knew there had to be a better solution than Pentair’s.  At another house we had a Hayward system that never worked very well either.  Not sure what it is about pool controllers.  They seem to be designed by very much “old school” electrical engineers (maybe) who seem to think ladder logic should and will take over the world.  I’m always impressed at how clunky, difficult to use, antiquated, & just plain crappy these commercial systems are.  And over priced?  Only by an order or two of magnitude.  I continue to be amazed that no one has come out with a truly modern, turn key, robust, elegant, & reasonably priced pool controller.  I know there are other products that are just as crappy in other domains, but c’mon folks!

I spent some time looking around the web & eventually found njs-PC.  I was intrigued by the possibility of using the ET as the hardware interface to the pool equipment but actually controlling things from HA.  

I had a spare Pi & installed njs-PC on it and used an RS-485 converter to send the bus communication over my Ethernet connection to the Pi located in my wiring closet.  I found @Crewski’s HA integration https://github.com/tagyoureit/nodejs-poolController/discussions/559 and figured I was in Fat City with the best of both worlds—easy setup for njs-PC, easy integration to HA, and minimal ET4 reliance with far superior information & control.  Woo hoo!

Eventually, and after some growing pains (I’m sure you can find my posts if you care), I moved njs-PC to my HA box that runs Docker. njs-PC was now another Docker compose container.  Along my journey, I learned how incredibly helpful folks associated with this github project are, especially @tagyoureit  and @rstrouse.   These guys not only created, nurture, improve, & support njs-PC but they’re amazingly responsive, knowledgeable, caring, & helpful.  True Rock Stars!!!

My implementation used HA to control all pool functionality with the ET as its slave.  I had far more information about the state of the pool and I blocked the ET’s ability to call out to the Mothership of Pentair.  I had remote access to everything via my locally-installed Wireguard VPN.  Good times!

Of course, as it turned out, there was a not-so-minor chink the the armor—the ET4 board itself.  My ET 4 board failed two more times due to RS-485 failures in 2023 & I was getting rather unhappy, to say the least.  My warranty was due to expire in October 2023 and I had no intention of becoming beholden to the local pool company or Pentair.  Of course, there’s a discussion on ET4 board repair within njs-PC https://github.com/tagyoureit/nodejs-poolController/discussions/619, as well as others on the web, and even YouTube videos on diagnosis and 485 chip replacement.   Apparently, Pentair uses among the junkiest 485 chips out there and their failure rate is legendary.  The ET boards appear to be profit center for Pentair, since they run $500.00 and up for a replacement, plus installation for the uninitiated.  

It seems the board design (about 20 years old) and specific 485 chip are highly sensitive to power fluctuations, especially those caused by lighting strikes, *even if your house is not struck by lightning or had a significant power surge!*  Funny that in spite have having hundreds of smart devices in my house, including WiFi, Zigbee, BLE, Z-wave, etc. none of them ever failed due to power surges.  But my “trusty” ET4 happily fried when the lights didn’t even flicker.  The ET 4 even propagated a surge that blew out 4 ports on my Ubiquiti switch.  Of course, nothing else on my network even blinked when the ET damaged my switch & then, its mission completed, the ET itself committed suicide.  As a result, I re-routed the ET Ethernet though a surge suppressor prior to plugging it into a working port on my switch.  Oh, what a world!!!

I read the Wiki on building your own Nixie a few times & was a bit overwhelmed by the project.  I’m not a hardware guy & the project includes messing with a fair chunk of hardware, at least for me.  I was also not thrilled by the prospect of messing around much with the high voltage portions of the pool equipment; it seemed a bit too far outside of my comfort zone.  

When the pool company installed yet another replacement board last summer, the installer made an error in its configuration that truly crossed the line.  The pool has an automated valve that controls an emergency drain line because we often get torrential rains that overwhelm the sole ½” overflow line installed in the pool.  Our house was designed without steps anywhere (other than the 4 into the pool) because we’re old coots & saw how our aging parents struggled to navigate even a single step with a 4” rise.  We were determined to make our house as wheelchair and walker-friendly as can be & hope we never need either.  Our overly optimistic plan is to remain as healthy as possible and stay in the house for the long run, despite whatever challenges may lie ahead.  Anyway, the pool has overflow sensors that—via Home Assistant—trigger the pump to start (if not already running) and drain line valve to open & drain the pool for 10 minutes & then see if another cycle is needed.  As a final backstop, the ET had a 10 minute egg timer on the valve so we don’t empty the pool if HA or other parts of the system go south.  Or so I thought...

Well, our pool technician erred when programming the shiny new ET board to open the drain valve when the pool started.  He also “forgot” to program the egg timer for the drain valve.  The next morning the pool turned on while we were walking our dogs.  When we returned home I heard some water flowing in the backyard.  The water level in the pool was down about 12”.  To my chagrin, the drain line was open, the pump was cruising away because the Polaris was on, & water was  pouring out through the drain line.  Ugh!  It was time for the local pool company to go and time for a new & different solution.

At this, my darkest (pool) hour, a White Knight appeared on the  horizon... @CodeSlinger69.  This fine fellow was converting his ET 4 into a Nixie while using as much as practicable of the existing hardware and he generously had the wherewithal to carefully document his experience!  https://github.com/tagyoureit/nodejs-poolController/discussions/801  The clouds began to part, the sun shined again, and the pool automation angels sang!  Perhaps, with careful study, some willingness to go  beyond my comfort zone, & a massive amount of help and patience from participants in these forums, I could complete a similar project successfully and safely.  It was time for a new adventure to begin...

**Comment by bwinstonbillville** [comment]
Great story - just starting my dive into this and I found this inspiring.  Thanks for sharing - and quite well written btw.

**Reply by FlaMike** [reply]
Thank you for your kind response!  You're inspiring me to get on with the next chapter...  design of my implementation.  I'll try to get cracking on it soon.  I hope it turns out to be useful to you & others.
